securedrop_client/gui/widgets.py

```diff
@@ -19,181 +19,553 @@
-        layout.setContentsMargins(20, 10, 20, 10)
+    def __init__(self):
+        super().__init__()
 
-        self.setAutoFillBackground(True)
-        palette = QPalette()
-        palette.setBrush(QPalette.Background, QBrush(load_image('hexes.svg')))
-        self.setPalette(palette)
+        # Set styles
+        self.setStyleSheet(self.css)
+        self.setFixedWidth(200)
+
+        # Set layout
+        layout = QHBoxLayout(self)
+        self.setLayout(layout)
+
+        # Remove margins
+        layout.setContentsMargins(0, 0, 0, 0)
+
+        # Login button
+        self.login_button = LoginButton()
+
+        # User icon
         self.user_icon = QLabel()
-        self.user_icon.setFont(QFont("Helvetica [Cronyx]", 16, QFont.Bold))
-        self.user_icon.hide()
-        self.user_state = QLabel()
-        self.user_state.setFont(QFont("Helvetica [Cronyx]", 12, QFont.Bold))
-        self.user_state.hide()
-        self.user_menu = JournalistMenuButton(self)
-        self.user_menu.hide()
-
-        self.login = QPushButton(_('SIGN IN'))
-        self.login.setFont(QFont("Helvetica [Cronyx]", 10))
-        self.login.setMinimumSize(200, 40)
-        button_palette = self.login.palette()
-        button_palette.setColor(QPalette.Button, QColor('#eee'))
-        button_palette.setColor(QPalette.ButtonText, QColor('#000'))
-        self.login.setAutoFillBackground(True)
-        self.login.setPalette(button_palette)
-        self.login.update()
-        self.login.clicked.connect(self.on_login_clicked)
+        self.user_icon.setObjectName('user_icon')
 
-        spacer = QWidget()
-        spacer.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
-
-        logo = QLabel()
-        logo.setMinimumSize(200, 200)
-        logo.setPixmap(load_image('logo.png'))
-
-        user_layout = QHBoxLayout()
-        user_layout.addWidget(self.user_icon, 5, Qt.AlignLeft)
-        user_layout.addWidget(self.user_state, 5, Qt.AlignLeft)
-        user_layout.addWidget(self.login, 5, Qt.AlignLeft)
-        user_layout.addWidget(self.user_menu, 5, Qt.AlignLeft)
-        user_layout.addStretch()
-
-        layout.addLayout(user_layout)
-        layout.addWidget(spacer, 5)
-        layout.addWidget(logo, 3, Qt.AlignCenter)
+        # User button
+        self.user_button = UserButton()
+
+        # Add widgets to user auth layout
+        layout.addWidget(self.login_button, 1)
+        layout.addWidget(self.user_icon, 1)
+        layout.addWidget(self.user_button, 4)
+
+        # Align content to the left
         layout.addStretch()
 
     def setup(self, window, controller):
-        """
-        Store a reference to the GUI window object (through which all wider GUI
-        state is controlled).
+        self.user_button.setup(controller)
+        self.login_button.setup(window)
+
+    def set_username(self, username):
+        self.user_icon.setText(_('jo'))
+        self.user_button.set_username(username)
+
+    def show(self):
```

Comment:
This is another case where the show/hide methods could possibly be replaced by parent/child + show/hide of the parent.